File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    a = brown + yellow
    for i in range(1, int(a**(1/2))+1):
        x, y = 0, 0
        if a%i==0:
            x = i
            y = a//i
            if (x-2)*(y-2) == yellow:
                answer.append(x)
                answer.append(y)
        
    return sorted(answer, reverse=True)

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let answer = [];
    

    let area = brown + yellow;
    
    for(let height=3; height<parseInt(area/2); height++){
        if((height-2) * (parseInt(area/height) - 2) === yellow){
            answer.push(parseInt(area/height));
            answer.push(height);
            break;
        }
    }
    
    return answer
}